### PR TITLE
[EAK-382] Fixed injecting child resources

### PR DESCRIPTION
**Pull request by anvovk**
[EAK-382]  Fixed injecting of nested child resources of 1st level. Now if a field class is annotated by @Child and contains inside Resource fields, these fields will be injected as well as other types of fields. But further injection of of nested elements deeper than 1st lvl is not supported.

### CODE REVIEW

**Comment by anvovk** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by vovkandrew** [comment]
I have read the CLA Document and I hereby sign the CLA

File: core/src/test/java/com/exadel/aem/toolkit/core/injectors/models/TestModelChildInner.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+
+import com.exadel.aem.toolkit.api.annotations.injectors.Child;
+
+@Model(adaptables = SlingHttpServletRequest.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
+public class TestModelChildInner {
+    @Child
+    private Resource innerModelResource;
+
+    @ValueMapValue
```
Comment (mhoroshko) [inline core/src/test/java/com/exadel/aem/toolkit/core/injectors/models/TestModelChildInner.java:16]: Can we check also usage with relative path? for example `@ValueMapValue(name="innerModelResource/title")`

File: core/src/test/java/com/exadel/aem/toolkit/core/injectors/models/TestModelChildOuter.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.models.annotations.DefaultInjectionStrategy;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+
+import com.exadel.aem.toolkit.api.annotations.injectors.Child;
+
+@Model(adaptables = SlingHttpServletRequest.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
+public class TestModelChildOuter {
```
Comment (mhoroshko) [inline core/src/test/java/com/exadel/aem/toolkit/core/injectors/models/TestModelChildOuter.java:11]: (cosmetic) Can we give it more specific name for example for outer: `TestModelChildRootResource` and for inner `TestModelChildSubResource`?

**Review by mhoroshko — APPROVED** [review]

**Pull request merged**